scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,246 @@
+    fi
+
+    # return the kubecconfig of the control plane in plain text
+    echo "$kubeconfig"
+}
+
+
+while (( $# > 0 )); do
+    case "$1" in
+    (--names|-n)
+        if (( $# > 1 ));
+        then { arg_names="$2"; shift; }
+        else { echo "$0: missing list of KubeFlex Control Plane names" >&2; exit 1; }
+        fi;;
+    (--kubeconfig|-k)
+        if (( $# > 1 ));
+        then { arg_kubeconfig="$2"; shift; }
+        else { echo "$0: missing kubeconfig filename" >&2; exit 1; }
+        fi;;
+    (--context|-c)
+        if (( $# > 1 ));
+        then { arg_context="$2"; shift; }
+        else { echo "$0: missing context name" >&2; exit 1; }
+        fi;;
+    (--replace-local-ip-address|-r)
+        if (( $# > 1 ));
+        then { arg_ip_addr="$2"; shift; }
+        else { echo "$0: missing ip address" >&2; exit 1; }
+        fi;;
+    (--merge|-m)
+        arg_merge=true;;
+    (--output|-o)
+        if (( $# > 1 ));
+        then { arg_output="$2"; shift; }
+        else { echo "$0: missing output filename" >&2; exit 1; }
+        fi;;
+    (--silent|-s)
+        arg_verbose=false;;
+    (-X)
+        set -x;;
+    (-h|--help)
+        display_help
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/KUBECONFIG="$in_KUBECONFIG" kubectl/generated/KUBECONFIG="$in_KUBECONFIG" kubectl_config/
+echov "Determining the list of kubeconfigs to search:"
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    in_KUBECONFIG="$arg_kubeconfig"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$arg_kubeconfig"
+    fi
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    in_KUBECONFIG="$KUBECONFIG"
+    if [[ "$arg_output" == "" ]] ; then
+        if [[ "$KUBECONFIG" == *":"* ]] ; then
+            out_kubeconfig="$HOME/.kube/config" # hard to decide
+        else
+            out_kubeconfig="$KUBECONFIG"
+        fi
+    fi
+else
+    in_KUBECONFIG="$HOME/.kube/config"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$HOME/.kube/config"
+    else
+        out_kubeconfig="$arg_output"
+    fi
+fi
+echov "- input kubeconfigs: \"$in_KUBECONFIG\""
+echov "- output kubeconfig: \"$out_kubeconfig\""
+
+
+# Determine the list of contexts to search
+echov "Determining the list of contexts to search:"
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(KUBECONFIG="$in_KUBECONFIG" kubectl config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+for j in "${!contexts[@]}" ; do # for all contexts
+    echov "- \"${contexts[j]}\""
```

Comment:
Maybe add "searching context:" in there?

Reply:
yes, I see that structure. My point is that the user gets a list of `- <something>` here without being told what those somethings are. I think that it would be better to output some explanation of what this list is.

Reply:
This is not a blocker.

Reply:
I added some text